### PR TITLE
Improve SQL logging readability

### PR DESCRIPTION
**Pull request by lzap**
This pull request improves the readability of SQL logs by formatting them in a more readable way. It also adds unit tests to ensure the correctness of the formatting function.

### CODE REVIEW

File: internal/db/db_trace.go [hunk 1]
```diff
@@ -11,10 +12,37 @@ import (
 type dbTracer struct{}
 
 func (dt *dbTracer) TraceQueryStart(ctx context.Context, conn *pgx.Conn, data pgx.TraceQueryStartData) context.Context {
-	logrus.WithContext(ctx).Debugf("Executing SQL with args %v: %s", data.Args, data.SQL)
+	logrus.Debug(formatSqlLog(data))
```
Comment (ezr-ondrej) [inline internal/db/db_trace.go:15]: don't we want to log to the context logger? :thinking:

File: internal/db/db_trace.go [hunk 2]
```diff
@@ -11,10 +12,37 @@ import (
...
+
+func formatSqlLog(data pgx.TraceQueryStartData) string {
+	if logrus.GetLevel() > logrus.DebugLevel {
+		return ""
+	}
+
+	d := make([]interface{}, len(data.Args))
+	copy(d, data.Args)
+	for i, v := range d {
+		if b, ok := v.([]byte); ok {
```
Comment (kingsleyzissou) [inline internal/db/db_trace.go:31]: So the thing it's not a `[]byte` it's a `json.RawMessage` :

I was messing around here:
https://github.com/osbuild/image-builder/commit/d31d61c7c773b898bfa324ad1c5f51aeebba8a51

Comment (mgold1234) [reply internal/db/db_trace.go:31]: @kingsleyzissou and I verified that the correct logs present  sudo docker compose logs backend | grep -i 'insert'  
image-builder-backend-1  | time="2024-03-21T12:22:53Z" level=debug msg="Executing SQL: \n\t\tINSERT INTO composes(job_id, request, created_at, account_number, email, org_id, image_name, client_id, blueprint_version_id)\n\t\tVALUES ($1, $2, CURRENT_TIMESTAMP, $3, $4, $5, $6, $7, $8); args: [03120b06-71a4-429a-bc40-03c9922272a1 {\"client_id\":\"ui\"... 6089719 insights-qe-stage-notifications+insights-qa@redhat.com 11789772 <nil> 0x4001792110 <nil>]" func="github.com/osbuild/image-builder/internal/db.(*dbTracer).TraceQueryStart" file="/go/src/github.com/osbuild/image-builder/internal/db/db_trace.go:16"

Comment (kingsleyzissou) [reply internal/db/db_trace.go:31]: The `json.RawMessage` contains the user email address etc, are we okay with logging all of that information?

File: internal/db/db_trace.go (hunk 1 above)
Comment (lzap) [reply internal/db/db_trace.go:15]: Good catch.

**Comment by lzap** [comment]
I have incorporated few more changes from my other PR, all are log improvements.

File: internal/db/db_trace.go (hunk 2 above)
Comment (kingsleyzissou) [reply internal/db/db_trace.go:31]: Makes sense. Yeah we will probably need to add a case for converting the `json.RawMessage` too in that case :)

Comment (lzap) [reply internal/db/db_trace.go:32]: Ah that is right, thanks! Added a test case for this too.

**Review by kingsleyzissou — APPROVED** [review]
Looks good, thanks!

**Pull request merged**